website_user_login_redirect_custom/tests/test_website_login_redirect.py

```
@@ -0,0 +1,136 @@
+            "http://evil.com",
+            "https://test.com",
+            "//phish.com",
+            "javascript:alert(1)",
+            "data:text/html;base64,xxx",
+            "vbscript:msgbox(1)",
+            "  /shop",  # leading spaces
+            "/shop  ",  # trailing spaces
+        ]
+        for url in invalid_urls:
+            with self.subTest(url=url):
+                with self.assertRaises(ValidationError):
```

Ruff (0.14.0)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)